site/_posts/2023-02-03-single-sign-on-laravel-fusionauth.md

```diff
@@ -0,0 +1,339 @@
+
+{% include _callout-tip.liquid content="If you cloned our [GitHub repository](https://github.com/FusionAuth/fusionauth-example-laravel-single-sign-on), you can just copy the contents from [laravel/routes.php](https://github.com/FusionAuth/fusionauth-example-laravel-single-sign-on/blob/main/laravel/routes.php)." %}
+
+```php
+<?php
+
+use App\Models\User;
+use Illuminate\Support\Facades\Auth;
+use Laravel\Socialite\Facades\Socialite;
+
+/**
+ * Responsible for redirecting the user to the FusionAuth login page
+ */
+Route::get('/auth/redirect', function () {
+    return Socialite::driver('fusionauth')->redirect();
+})->name('login');
+
+/**
+ * This is the address that we configured in our .env file which the user will be redirected to after completing the
+ * login process
+ */
+Route::get('/auth/callback', function () {
+    /** @var \SocialiteProviders\Manager\OAuth2\User $user */
+    $user = Socialite::driver('fusionauth')->user();
+
+    // Let's create a new entry in our users table (or update if it already exists) with some information from the user
+    $user = User::updateOrCreate([
+        'fusionauth_id' => $user->id,
+    ], [
+        'name' => $user->name,
+        'email' => $user->email,
+        'fusionauth_access_token' => $user->token,
+        'fusionauth_refresh_token' => $user->refreshToken,
+    ]);
+
+    // Logging the user in
+    Auth::login($user);
+
+    // Here, you should redirect to your app's authenticated pages (e.g. the user dashboard)
+    return redirect('/');
+});
+```
+
+### Changing the view
+
+To better see what is going on after logging in, you can change the view for the home page to actually display the name for the current user. You can do this by opening `resources/views/welcome.blade.php` and adding a new line after "Home":
+
+```php
+// Before:
+@auth
+  <a href="{% raw %}{{ url('/home') }}{% endraw %}" class="text-sm text-gray-700 dark:text-gray-500 underline">Home</a>
+@else
+  // ...
+
+// After:
+@auth
+  <a href="{% raw %}{{ url('/home') }}{% endraw %}" class="text-sm text-gray-700 dark:text-gray-500 underline">Home</a>
+  <span class="ml-2 text-sm text-gray-700 dark:text-gray-500">{% raw %}{{ Auth::user()->name }}{% endraw %}</span>
+@else
+  // ...
+```
+
+## Testing
+
+It is finally time to open your application in the browser and test everything! Navigate to [http://localhost](http://localhost) and click the "Log in" link in the upper right corner of that page to be taken to FusionAuth's login screen. Provide the credentials for the user you created earlier and submit the form: you should be redirected back to that "Welcome" page but instead of seeing the same "Log in" link, you should now see both "Home" and your name there!
+
+{% include _image.liquid src="/assets/img/blogs/laravel-single-sign-on/laravel-application-logged-in.png" alt="The default Laravel application while being logged in." class="img-fluid" figure=true %}
```

Comment:
It was really hard to see the change here (it's in the upper right hand corner, in gray). Can you please highlight it with a red box with line weight 5)